proof_of_concept/local_workflow_runner.py

```diff
@@ -161,19 +166,24 @@ def _get_step_inputs(
             print('Job at {} getting input {} from site {}'.format(
                 self._this_runner, data_key, source_store))
             try:
-                step_input_data[inp_name], metadata = (
-                        self._ddm_client.retrieve_data(
-                            source_store, data_key))
+                asset = self._ddm_client.retrieve_asset(source_store, data_key)
+                step_input_data[inp_name] = asset.data
                 print('Job at {} found input {} available.'.format(
                     self._this_runner, data_key))
-                print('Metadata: {}'.format(metadata))
+                print('Metadata: {}'.format(asset.metadata))
             except KeyError:
                 print('Job at {} found input {} not yet available.'.format(
                     self._this_runner, data_key))
                 return None
 
         return step_input_data
 
+    def _retrieve_compute_asset(self, compute_asset_id: str) -> ComputeAsset:
```

Comment:
I'm getting: 185: error: Incompatible return value type (got "Asset", expected "ComputeAsset") from mypy here. I have no clue how to tell python that the store can have any subclass of Asset, but that I am expecting a ComputeAsset here (which might be a shortcoming of my current implementation). @LourensVeen let's discuss this!

Reply:
If you check this using `isinstance()`, then mypy is smart enough to deduce for each branch that it is the checked-for type. If it's not a ComputeAsset, we should raise, because someone submitted an invalid workflow.

Reply:
Fixed!